lib/helpers.js

```diff
@@ -24,6 +24,11 @@ function finallyRejectsWithInitial(promise, fn) {
 
 const promiseUtils = require('promise-callbacks');
 
+//  A simple way to get up to 14 random hex digits, not cryptographic
+const randomHex14 = () => {
+  return Math.floor(Math.random() * Number.MAX_SAFE_INTEGER).toString(16);
```

Comment:
We could prefix this string with the value of `Date.now()` so as to have IDs be ordered, if we believe this undocumented feature of existing IDs should be supported.  E.g. `1587387486767-10ef4cf53faaf`

Reply:
It's specifically problematic for us because we use it to compute influx metrics - every ten seconds we pull `checkHealth` and subtract the previous `id` from the new value.

Reply:
Hmmm, this PR doesn't affect the growth of `newestJob` -- specifically that semantic is preserved.  So, subtracting values and dividing by elapsed time will still give a throughput metric.

What is changed is that `newestJob` it is no longer an ID that's been in the system;  it is only there to be the counter (so arguably the name is wrong).  We could add a correct name, `jobCount` and deprecate the `newestJob` name...

I was talking about changing the ID to have a timestamp prefix solely to ensure that the IDs are lexically monotonic.

Reply:
>Hmmm, this PR doesn't affect the growth of `newestJob` -- specifically that semantic is preserved.

I'm not sure I understand how this works, then. The timestamp prefix wouldn't let us compute the delta between two `id`s to get a throughput metric - it would result in the amount of time elapsed instead of the number of jobs created - right?

>And, can we revisit why non-default-IDs (client IDs) are excluded from the count...?

Yes, that'd be fantastic! Not by design, just never prioritized.

Reply:
I think I understand.  

I had assumed that your throughput was based on taking two `checkHealth().newestJob` measures -- their difference is the number of jobs delta, and the difference in `Date.now()` between those two measures gets the time difference-- that's how I'm computing job throughput, as per hints in README.  Any Queue instance can make such a throughput determination and it doesn't need to handle any jobs to do so.

You are using the difference between two specific `job.id` to get the jobs delta, and also using difference between `Date.now()` for the time difference.  You have sentinel jobs (or something like that) and they are the ones that go into making the jobs-count delta.  Do understand correctly about using `job.id` and about specific jobs being used for throughput?

My discussion about a prefix on the `job.id` was solely an attempt to preserve the semantic that the existing `job.id` strings have a lexical ordering by virtue of coming from the incrementing counter.

Reply:
My bias, based on various experiences, is to avoid putting any semantics on identifier values as it's one form of lava that is particularly hard to break free of...